cdc/streamer.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdc
+
+import (
+	"time"
+
+	"github.com/apple/foundationdb/bindings/go/src/fdb"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/rs/zerolog/log"
+	"github.com/tigrisdata/tigrisdb/internal"
+	"github.com/tigrisdata/tigrisdb/server/config"
+)
+
+type Streamer struct {
+	db      fdb.Database
+	lastKey fdb.Key
+	cfg     config.CdcConfig
+	pub     *Publisher
+	ticker  *time.Ticker
+	Txs     chan Tx
+}
+
+func (s *Streamer) start() error {
+	key, err := s.db.ReadTransact(func(rtx fdb.ReadTransaction) (interface{}, error) {
+		begin := fdb.FirstGreaterThan(s.pub.beginKey)
+		end := fdb.LastLessThan(s.pub.endKey)
+		r := rtx.GetRange(fdb.SelectorRange{Begin: begin, End: end}, fdb.RangeOptions{Limit: 1, Reverse: true})
+
+		i := r.Iterator()
+		if i.Advance() {
+			kv, err := i.Get()
+			if err != nil {
+				return nil, err
+			}
+			return kv.Key, nil
+		} else {
+			return s.pub.beginKey, nil
+		}
+	})
+	if err != nil {
+		return err
+	}
+
+	s.lastKey = key.(fdb.Key)
+	s.Txs = make(chan Tx, s.cfg.StreamBuffer)
+	s.ticker = time.NewTicker(s.cfg.StreamInterval)
+	go func() {
+		for range s.ticker.C {
+			err := s.read()
```

Comment:
nit, combine it in one line `if err := s.read(); err != nil {}`

Reply:
k